encoding/bytesutil/hex.go

```diff
@@ -22,14 +22,10 @@ func IsHex(b []byte) bool {
 // DecodeHexWithLength takes a string and a length in bytes,
 // and validates whether the string is a hex and has the correct length.
 func DecodeHexWithLength(s string, length int) ([]byte, error) {
-	bytes, err := hexutil.Decode(s)
```

Comment:
This function was decoding without verifying the user input before.

Reply:
I actually think this is a slight loss of functionality. If you look at the functions this calls, the validation you are adding was already performed at different layers. The main change is that we lose the clear error message from the geth func when the hex string is missing the `0x` prefix.

Reply:
I think the problem is that there is a copy in memory of an arbitrary length buffer and I dislike this by default without a check before the copy

Reply:
That's fair!

Reply:
Possible compromise to keep existing error handling while adding your check?
```
func DecodeHexWithLength(s string, length int) ([]byte, error) {
        if len(s) > 2*length+2 {
                return nil, fmt.Errorf("%s is greater than length %d bytes", s, length)
        }
	bytes, err := hexutil.Decode(s)
	if err != nil {
		return nil, errors.Wrap(err, fmt.Sprintf("%s is not a valid hex", s))
	}
	if len(bytes) != length {
	        return nil, fmt.Errorf("%s is not length %d bytes", s, length)
	}
	return bytes, nil
}
```